.github/workflows/publish.yml

```diff
@@ -7,28 +7,46 @@ on:
 jobs:
   # publish the nuget package
   publish:
-    name: build, pack & publish
-    runs-on: ubuntu-latest
+    name: Build, Pack & Publish
+    runs-on: ${{ matrix.os }}
+    
+    strategy:
+      matrix:
+        os: [ ubuntu-latest, windows-latest ]
```

Comment:
With multiple values in the matrix the workflow steps below are executed twice, it looks like that we publish twice (o one for each value in the matrix). 
Remove the matrix (maybe os compatibility can be checked in the build workflows) or create a different step (independent, no matrix) for the actual Nuget publish

Reply:
Good catch